### PR TITLE
Edits to target packages to make intrinsics accessible in regenerated library code

### PR DESCRIPTION
**Pull request by bettinaheim**
Note that I did not make anything public that is part of the Q# Core, only the callables that are used in the target packages that replace it. This should hence be a non-breaking change. Unfortunately, having callables in target packages be internal makes it inpossible to regenerate the code that calls into it, and hence poses a problem.

### CODE REVIEW

**Comment by swernli** [comment]
I don't understand the problem this change is trying to solve, but I would be opposed to merging a change like this into main. Since this is just merging into a working branch it should be fine, so we can defer that discussion to a later date.

**Comment by bettinaheim** [comment]
> I don't understand the problem this change is trying to solve, but I would be opposed to merging a change like this into main. Since this is just merging into a working branch it should be fine, so we can defer that discussion to a later date.

@swernli The issue is that I need to instantiate the callables in the code that is regenerated. I may have missed something in how this was intended to work, but the C# generation as written was incompatible with using the target packages. I hence made the necessary update to have it in a working state. As written, I need to be able to instantiate the class that implements each gate as defined in the target package, hence the internal is interfering with that.

**Comment by swernli** [comment]
Could this just be an issue of regeneration of C# callables not respecting the `GenerateConcreteIntrinsics` property? C# generation with the target packages works in this repo when the target packages are testing with the simulator (see for example https://github.com/microsoft/qsharp-runtime/tree/main/src/Simulation/Simulators.Type1.Tests and corresponding test projects for the other target packages). We've run into quirks with internal intrinsics before (#389) and the concrete intrinsics feature was meant to solve this.

**Comment by bettinaheim** [comment]
> Could this just be an issue of regeneration of C# callables not respecting the `GenerateConcreteIntrinsics` property? C# generation with the target packages works in this repo when the target packages are testing with the simulator (see for example https://github.com/microsoft/qsharp-runtime/tree/main/src/Simulation/Simulators.Type1.Tests and corresponding test projects for the other target packages). We've run into quirks with internal intrinsics before (#389) and the concrete intrinsics feature was meant to solve this.

Setting GenerateConcreteIntrinsic to true in the corresponding project file (i.e. in QCIExe and in QSharpExe) did not solve the issue. You can try it out with package version 0.24.41904-alpha. However, there were a couple of issues I had to fix across repos, and another change in 0.24.41904-alpha may be interfering, so it may have been a case of it getting lost in the weeds. I gave it another try, now that I know what *all* the changes and issues were that needed to be solved: https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=41944&view=results

File: src/Simulation/Simulators.Tests/TestProjects/TargetedExe/TargetedExe.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
-    <QscVerbosity>detailed</QscVerbosity>
+	<QscVerbosity>detailed</QscVerbosity>
```
Comment (bettinaheim) [inline src/Simulation/Simulators.Tests/TestProjects/TargetedExe/TargetedExe.csproj:5]:
```suggestion
    <QscVerbosity>detailed</QscVerbosity>
```

**Pull request merged**